h2/src/main/org/h2/command/dml/RunScriptCommand.java

```diff
@@ -68,7 +69,20 @@ public long update() {
                 if (sql == null) {
                     break;
                 }
-                execute(sql);
+
+                // Normalize the statement
+                String upper = StringUtils.toUpperAscii(sql);
+                boolean stale = false;
+                for (String pattern : RunScript.STALE_OBJECT_PATTERNS) {
+                    if (upper.indexOf(pattern) >= 0) {
+                        stale = true;
+                        break; // stop early on first match
+                    }
+                }
+                if (!stale) {
+                    execute(sql); // keep original sql
```

Comment:
I don't see how you detect attempts to execute the same command more than one time.

Reply:
Sorry, I don't understand this comment. Please elaborate.
The SQL is just ignored and discarded w/o execution (which is correct since it should not even be there).

The correct way would be to avoid exporting/writing those 3 objects into the SQL dump but I have not found a good reliable way to understand how MVStore is dumped into SQL file.

Also, once you have a tainted SQL dump its pretty hard to remove those statements because the `ALTER TABLE ...` spans multiple lines that can't be removed by `sed` directly.

That's why I chose this very ugly, but robust approach.
It saved me 2 production DBs already.